### PR TITLE
527 EZ Pad Boxes with 0's

### PR DESCRIPTION
**Pull request by ATeal**
## Description of change
On 527 EZ form it requires that all of the financial fields be filled out. These changes ensure that happens

resolves https://github.com/department-of-veterans-affairs/vets-contrib/issues/830

## Testing done
- Rspec

## Testing planned
- End to end testing in january

## Acceptance Criteria (Definition of Done)
- All fields in the financial section are filled regardless of the data received.

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)

### CODE REVIEW

**Comment by lihanli** [comment]
was this change requested by someone at the VA? i'd assume the person reading the form would understand if the "other income expected" field was left blank there was nothing to report. also do we know if they understand what "recipient: none" means? in the instructions it says "Identify
the specific income recipient for each income source, yourself or another person in your household"

**Comment by kfrz** [comment]
LGTM. These form builders are quite large to parse, but your changes all make sense to me.

File: lib/pdf_fill/forms/va21p527ez.rb [hunk 1]
```diff
@@ -728,13 +730,11 @@ def expand_financial_acct(recipient, financial_acct, financial_accts)
...
         financial_accts.each do |acct_type, accts|
-          if accts.size.zero? && acct_type != 'additionalSources'
-            accts << {
-              'recipient' => 'Myself',
-              'amount' => 0
-            }
-          end
+          accts << DEFAULT_FINANCIAL_ACCT if accts.size.zero?
+          pad_accts = multi_line_accts.include?(acct_type) && accts.size < 2
+          (2 - accts.size).times { |_| accts << DEFAULT_FINANCIAL_ACCT } if pad_accts
```
Comment (SMLuthi) [inline lib/pdf_fill/forms/va21p527ez.rb]: Can a social security or salary acct type have MORE than two lines?

Comment (SMLuthi) [reply lib/pdf_fill/forms/va21p527ez.rb]: If Im reading this correctly then you're just trying to pad a default account unless its either socialSecurity/salary, then you are trying to pad it twice as they are represented by two lines. If thats the case than the iterator is probably over kill and can be rewritten a little cleaner? Currently its a little hard to discern what its trying to do (for someone who hasnt worked on this).

Comment (ATeal) [reply lib/pdf_fill/forms/va21p527ez.rb]: That's correct and agreed, I will clean this up a bit! Thanks for the feedback!

Comment (ATeal) [reply lib/pdf_fill/forms/va21p527ez.rb]: @SMLuthi Could you take another peek and see if it makes more sense now?

**Comment by ATeal** [comment]
> was this change requested by someone at the VA? i'd assume the person reading the form would understand if the "other income expected" field was left blank there was nothing to report. also do we know if they understand what "recipient: none" means? in the instructions it says "Identify
> the specific income recipient for each income source, yourself or another person in your household"

@lihanli The change was requested by one of the claims processors (@Lenora2 can confirm who if need be) 

I think the reason they have a problem is the "DO NOT LEAVE ANY ITEMS BLANK" portion that's triggering more work for them

File: lib/pdf_fill/forms/va21p527ez.rb [hunk 2]
```diff
@@ -727,14 +729,14 @@ def expand_financial_acct(recipient, financial_acct, financial_accts)
...
+
       def zero_financial_accts(financial_accts)
         financial_accts.each do |acct_type, accts|
-          if accts.size.zero? && acct_type != 'additionalSources'
-            accts << {
-              'recipient' => 'Myself',
-              'amount' => 0
-            }
-          end
+          accts << DEFAULT_FINANCIAL_ACCT unless accts.any?
```
Comment (lihanli) [inline lib/pdf_fill/forms/va21p527ez.rb]: are you trying to check for the empty array here with `accts.any?`? how about using `accts.present?`, `.any?` isn't actually checking for an empty array

Comment (ATeal) [reply lib/pdf_fill/forms/va21p527ez.rb]: Had a fun read about `.any?` vs `.present?` vs `.blank?` vs `.empty?` I'm cool with `.present?` I'll update it

**Comment by ATeal** [comment]
@rtluu This one may have been review by the proper people already just FYI

**Pull request merged**